Core/wbDefinitionsSF1.pas

```diff
@@ -9209,15 +9222,21 @@ procedure DefineSF1b;
         ], []),
         //BGSExternalComponentDataSource_Component
         wbRStruct('Component Data', [
-          wbFormID(EXDC),
-          wbUnknown(EXDZ),
-          wbRStructs('Unknown', 'Unknown', [
-            wbString(EXCN),
-            wbFormID(EXCI)
+          wbFormID(EXDC, 'External Base Template'),
+          wbRStruct('External Data Sources', [
+            wbInteger(EXDZ, 'Data Source Count', itU32), // count for EXCN/EXCI struct array
+            wbRArray('External Sources',
+              wbRStruct('Component', [
+                wbString(EXCN, 'Component Name'),
+                wbFormID(EXCI, 'Component Source')
+              ], [])
+            )
           ], []),
-          wbUnknown(EXAC),
-          wbRArray('Unknown', wbString(EXAS)),
-          wbUnknown(EXBS)
+        wbRstruct('Unknown', [
+          wbInteger(EXAC, 'Count', itU32), // count for EXAS array
+          wbRArray('Unknown', wbString(EXAS), cpNormal, False, nil, wbEXASsAfterSet)
+        ], []),
+          wbString(EXBS).SetRequired(True)
```

Comment:
something looks wrong with the indendation here?

Reply:
yes